.github/workflows/security-scan.yml

```diff
@@ -0,0 +1,79 @@
+name: Security Scan
+
+on:
+  push:
+    branches:
+      - main
+      - release/**
+    paths-ignore:
+      - 'README.md'
+      - 'CHANGELOG.md'
+      - '.changelog/**'
+      - '.tours/**'
+      - 'contributing/**'
+  pull_request:
+    branches:
+      - main
+      - release/**
+    paths-ignore:
+      - 'README.md'
+      - 'CHANGELOG.md'
+      - '.changelog/**'
+      - '.tours/**'
+      - 'contributing/**'
+
+# cancel existing runs of the same workflow on the same ref
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  scan:
+    runs-on: custom-linux-xl-nomad-22.04
```

Comment:
If we're planning on running this in the ENT repo, we may want to do something like this: https://github.com/hashicorp/nomad/blob/main/.github/workflows/test-core.yaml#L52

Reply:
https://github.com/hashicorp/nomad/pull/20465/commits/0c2d20ed6347cbe1b780d858a578be2c5a9ee885